Gems/AtomLyIntegration/CommonFeatures/Code/Source/Mesh/EditorMeshComponent.cpp

```diff
@@ -253,6 +254,12 @@ namespace AZ
             AzToolsFramework::ToolsApplicationEvents::Bus::Broadcast(
                 &AzToolsFramework::ToolsApplicationEvents::InvalidatePropertyDisplay,
                 AzToolsFramework::Refresh_EntireTree);
+
+            // Register this component with the editor mode feedback system
+            if (auto* editorModeFeedbackInterface = AZ::Interface<EditorModeFeedbackInterface>::Get())
+            {
```

Comment:
Instead of be called from this method perhaps you can register to the event and have a callback method?

Reply:
Actually I'm going to close this one, as you require to register the entity and this will not be possible via OnModelReady (as far as I see)

Reply:
Looking into it, you won't get the EntityId and that's a a problem in your case, however, you can create another event for the (Editor)MeshComponent to announce when an entity model is ready and now you can publish this event here and listen to it in your class.

Reply:
I've looked into this at some depth now and I think the way to go would be something like this:

* A new `MeshComponentNotifications` event that `EditorEditorModeFeedbackSystemComponent` can subscribe to in order to be notified when a given entity's mesh is updated (the mesh handle will be passed along to the subscriber via this event).
* The `EditorMeshComponent` publishes said event when the model changes.
* The `EditorEditorModeFeedbackSystemComponent` subscribes to `MeshComponentNotifications` for each entity in the prefab when Focus Mode is activated, whereupon the mesh handle will be retrieved upon connection (should it already exist) and the `MeshDrawPacket`s constructed.
* Crucially, a prefab notification bus is implemented/augmented to allow subscribers to be informed when an entity is added/removed from the focused prefab such that the `MeshDrawPacket`s can be constructed/removed accordingly (@AMZN-daimini) as currently, using `OnEntityInfoUpdatedRemoveChildEnd` there isn't a reliable way to infer at the time of the notification publication which entity is actually being removed from the prefab.

This approach solves two problems: firstly, `EditorMeshComponent`'s coupling to `EditorModeFeedbackInterface` is removed and secondly, the construction/destruction of `MeshDrawPacket`s will now happen in response to focus mode prefab editing, not something that will occur in the `OnTick` handler. 

However, this is quite a sizeable change to the PR and will require a contribution from the Editor team so I think it's best if I make a note of this discourse and carry out the changes in a subsequent PR as to not shift the focus of this PR too much.

Reply:
I think that the first 3 points are very relevant to this system and will probably not create very large code change (added single callback as you wrote and your new component listens to that). 
I don't see the immediate affect/connection to prefabs - this exists / done indirectly and either exists already or not.

Here is what I suggest, try implementing this single callback and listen to it. If you find out that it takes more than several hours of effort, go back to your original implementation - I suspect that you find it rather simple once you dive into it.